geps/gep-2659.md

```diff
@@ -0,0 +1,199 @@
+
+Additionally, we've had two GEPs moved to Declined ([GEP-735: TCP and UDP address
+matching](https://gateway-api.sigs.k8s.io/geps/gep-735/) and
+[GEP-1282: Describing Backend Properties](https://gateway-api.sigs.k8s.io/geps/gep-1282/)).
+In the case of GEP-1282, we now have a replacement GEP,
+[GEP-1897: BackendTLSPolicy](https://gateway-api.sigs.k8s.io/geps/gep-1897/)
+which obsoletes the older GEP. But we have no way of representing this
+relationship (or any other, in fact) between GEPs at the moment.
+
+With these previous two changes, the addition of a metadata YAML file, similar
+to what the KEP process uses, seems like it is increasingly necessary. This GEP
+introduces a schema, which will be detailed in `.go` files rather than completely
+in this document.
+
+Lastly, I think that we should clarify our use of [RFC2119](https://www.rfc-editor.org/rfc/rfc2119.txt)
+language - we use MUST, SHOULD, MAY and so on as per that RFC in general,
+but there is an extension in [RFC8174](https://www.rfc-editor.org/rfc/rfc8174.txt)
+that adds that these words are only to be interpreted per the RFC when they are
+in ALL CAPS, and "must" "should", "may" are to be interpreted in their usual
+English meaning (which is not as strong as the RFC2119 one). This seems like
+a _very_ good idea to adopt to me.
+
+## Proposals
+
+### Adopt the RFC8174 modification to RFC2119 language
+
+RFC8174 clarifies that the reserved words MUST be in all-caps to have their
+assigned meaning. This should make the spec clearer over time as we migrate.
+
+### Addition of the concept of GEP "types", and a new GEP Type
+
+We will introduce the concept of GEP types.
+
+#### Types of GEP
+There will be two types of GEP. The existing type will be called `API` GEPs, and
+a new type `Memorandum` (after "Memorandum of Understanding") will be created.
+
+##### API GEPs
+
+GEPs that do not have an assigned type will be considered `API` GEPs. (For
+legacy GEPs). New GEPs created after this GEP is merged will default to the `API`
+type.
+
+As they do currently, these GEPs lay out a problem the API needs to address,
+along with a rationale and a solution for that problem.
+
+This also uses the existing GEP process and phases:
+
+1. Discuss
+1. Provisional
+  1. (optional) Prototyping
+1. Implementable
+1. Experimental
+1. Standard
+1. Completed
+
+As part of the addition of relationships, we will extend these phases, but see
+the "Addition of GEP relationships" section for details, but in short, when an
+API GEP has been Obsoleted, it will be moved to a new "Obsolete" phase (which
+is functionally the same as "Declined").
+
+##### Memorandum GEPs
+
+These GEPs record an agreement on either the definition of a problem, its scope
+and a common language, or further process changes to the GEP process itself.
+The defining characteristic here is that the GEP MUST NOT result in any changes
+to the Gateway API spec, and SHOULD result in further GEPs to further clarify.
```

Comment:
It seems like at least some of these GEPs aren't going to require follow up. For example, we have a GEP to define how Gateway API interacts with Multi-Cluster Services. I think that could _potentially_ make a case for being a Memorandum because it's not making any API changes. At the same time, I don't think that GEP will really need any follow-up GEPs. Perhaps some updates to the existing GEP, but I think that would be it.

Reply:
Agreed, that's why this was a SHOULD, but I guess I could downgrade to MAY?